.github/actions/install-pre-commit/action.yml

```diff
@@ -19,31 +19,37 @@
 name: 'Install pre-commit'
 description: 'Installs pre-commit and related packages'
 inputs:
+  # TODO(potiuk): automate update of these versions
   python-version:
     description: 'Python version to use'
-    default: 3.9
+    default: "3.9"
   uv-version:
     description: 'uv version to use'
-    default: 0.5.5
+    default: "0.5.13"
   pre-commit-version:
     description: 'pre-commit version to use'
-    default: 4.0.1
+    default: "4.0.1"
   pre-commit-uv-version:
     description: 'pre-commit-uv version to use'
-    default: 4.1.4
+    default: "4.1.4"
 runs:
   using: "composite"
   steps:
+    - name: "Restore pre-commit[pre-commit] cache"
+      uses: apache/infrastructure-actions/stash/restore@c94b890bbedc2fc61466d28e6bd9966bc6c6643c
+      with:
+        # yamllint disable rule:line-length
+        key: "pre-commit-pre-commit-cache-${{ inputs.python-version }}-${{ hashFiles('.pre-commit-config.yaml') }}"
+        path: ~/.cache/pre-commit
+    - name: "Restore pre-commit[uv] cache"
+      uses: apache/infrastructure-actions/stash/restore@c94b890bbedc2fc61466d28e6bd9966bc6c6643c
```

Comment:
Can we point these without sha? as it is from our internal apache repo so we dont need sha?

Reply:
This is a bit of a convention we need to work out when we share multiple actions in one repo - that more difficult when there is one action per-repo, because the tag is "per-repo". Somehow I like the idea that we have several actions in "infrastructure-actions" (i.e. `monorepo`) - but then we need to figure out a convention for tag name we  will have to use. I think the convention we have in Airlfow (providers-fab/1.2.0) is a good one, maybe we can follow similar `stash/v1` ? I will propose it on slack.

Reply:
agreed stash/v1 would be better. lets see how it goes in slack :)